src/detect-parse.c

```diff
@@ -855,14 +856,28 @@ int SigParse(DetectEngineCtx *de_ctx, Signature *s, char *sigstr, uint8_t addrs_
          * output variable. */
         do {
             memset(output, 0x00, buffer_size);
+            plist = s->list;
             ret = SigParseOptions(de_ctx, s, input, output, buffer_size);
             if (ret == 1) {
+                /* Check for a transition out of a terminal state. */
```

Comment:
I think I would prefer having this check go elsewhere, like in SigValidate. It runs post-parsing, but it keeps the core parsing function cleaner.